### DISCUSSION TITLE
BangleJs vs PineTime sample size 1

### DISCUSSION THREAD

**Discussion by lundstrj** [OpenSeizureDetector/Android_Pebble_SD#163] [04_Devices]
I've been using PineTimeSD 0.6 for about a week before switching back to the Bangle again.

It seems to me that the PineTime offers a significantly better HRM and a pretty snappy experience but perhaps at the expense of Bluetooth connection stability. Due to connectivity problems (with just a wall and maybe a distance of 1.5m between the watch and the phone, I kept experiencing connectivity problems with the PineTime).

Switching to the Bangle meant that the HRM graph didn't look as nice when I looked in the morning but the night had passed without connectivity problems and not a single warning nor alarm beep (in contrast to the nights with the PineTime where OSD would start warning but without indicating anything being wrong in the UI and then sometimes also sounding the alarm due to Fault).

I am pretty fortunate to be able to compare different devices like this (VenuSQ, PineTime, Bangle). Who else has had a chance to compare devices and what have you discovered?

**Comment by jones139** [comment]
I have quite a collection of watches :).

Your point about PineTime connection reliability is an interesting one because when I decided to concentrate on PineTime rather than BangleJS, the reliability seemed similar (both very good).    I have just been testing my original PineTime watch again and it has been running on a low spec phone for over 48 hours without disonnnecting.

But I also have a second batch of 3 PineTimes, and I have only taken one out of the box.   I think I have seen the disconnection issue that you, and another couple of users have reported.  So I am wondering if all PineTime watches are not equal.   I have another watch from my first batch which I have taken the back off previously - at the weekend I will test that, and if that shows good reliability and my newer one still does not, I might do some surgery to look for the differences.....

**Reply by jones139** [reply]
I should have said - I am working on an auto-reconnect function for the Android App which I am hoping will resolve the PineTime connectivity issue.

**Reply by lundstrj** [reply]
The PineTime is a pretty cheap, low spec device. It has also been around for years. I fully expect them to have done minor revisions over the years and potentially replaced components (which shouldn't matter but which might make a difference anyway).

Not to mention that not every device coming off an assembly line is as perfect as intended.

I do have another PineTime (likely from the same production run) which I can try. I just need to convince my other son to give it up (he likes to rack up a high step count).

**Reply by jones139** [reply]
I have made a draft version 0.7 of the PineTime Firmware (https://github.com/OpenSeizureDetector/PineTimeSD/releases/tag/v0.7) which attempts to set the bluetooth power to maximum. 
I can't quite decide if it makes any difference or not, so thought I'd release it in case users can decide.

**Comment by jones139** [comment]
I just spotted this page (https://www.joaquim.org/pinetime-upgrade-board/) - @joaquimorg has done a lot of work to design a replacement board for the PineTime.     
We could build on that work to do a slight modification to add some extra sensors - I am thinking that a thermistor to measure (close to) skin temperature might be an interesting (and easy) thing to add into it.   Skin Conductivity would be interesting too, but it would not be any use to me - our Benjamin's skin is too sensitive to have metal electrodes in contact with it.

**Comment by pmithrandir** [comment]
Hi,

We would be a company, I would underline the potential of this project as R&D, with a probability of being successful to be measured.

We are talking about a new board with still a lot of questions: 
 - Who will produce it
 - What about the fimware dev
 - Is it worth dropping the official pinetime support ?
 - is it stable / reliable in time for almost a full time usage
 - What about the battery consumption ? (more cpu usually means more energy needed)

We have a bit to gain with that project for sure, but the cost in hours invested is not small.
And we are getting far from the initial goal : To have a pretty decent seizure detection system affordable and easy to put in place.

My impression would be that although that part seems quite fun, I don't see it making us closer to our goal soon.

**Reply by jones139** [reply]
You are right - I had looked at making my own device a while back, but gave up when I realised that PineTime was about 85% of what I wanted - I only got to the breadboard stage
If we did it, I would make it PineTime compatible (this one is a replacement board, so no need to design a new case etc) - but you are right that to get it further than a prototype would be a huge amount of work - I really just wanted to record that page somewhere so I can find it again in the future.
